File: 04_recursion/recursion_solution/PrintOddEvenSequence/solution.js
function printOddDecreasing(n) {
	// Base case
	if (n < 1) return;

	// Check if it is odd
	if (n % 2 != 0) {
		console.log(n);
	}

	printOddDecreasing(n - 1);
}

function printEvenIncreasing(i, n) {
	// Base case
	if (i > n) {
		return;
	}

	// Check if i is even
	if (i % 2 == 0) {
		console.log(i);
	}

	printEvenIncreasing(i + 1, n);
}

function printSequence(num) {
	printOddDecreasing(num);
	printEvenIncreasing(2, num);
}

printSequence(6);

File: 04_recursion/recursion_solution/PrintOddEvenSequence/Solution.java
package PrintOddEvenSequence;

public class Solution {
    public static void printSequence(int num) {
		printOddSequence(num);
		printEvenSequence(2, num);
	}	

	public static void printOddSequence(int num) {
		if(num <= 0) {
			return;
		}

        // Doing this to skip the uneccessary checking of even numbers
		if(num % 2 == 0) {
			printOddSequence(num - 1);
		} else {
			System.out.print(num + " ");
			printOddSequence(num - 2);
		}
	}

	public static void printEvenSequence(int num, int limit) {
		if(num > limit) {
			return;
		}

        // Doing this to skip the uneccessary checking of odd numbers
		if(num % 2 != 0) {
			printEvenSequence(num+1, limit);
		} else {
			System.out.print(num + " ");
			printEvenSequence(num+1, limit);
		}
	}

}
